src/main/java/ch/uzh/ifi/hase/soprafs22/service/UserService.java

```diff
@@ -882,4 +882,25 @@ public void blockUser(long userId, long otherUserId) {
     userRepository.saveAndFlush(user);
     userRepository.saveAndFlush(otherUser);
   }
+
+    /**
+     * recursive function to delete all past games without plays on it
+     * @param user user where games have to be deleted
+     */
+    public void deleteAllEmptyPastGames(User user) {
```

Comment:
method name does not correspond with what it does: recursion stops as soon as it encounters a game with plays, but there can be newer games with no plays that don't get deleted.